pandas/core/window/rolling.py

```diff
@@ -1186,13 +1186,18 @@ def count(self):
         window = self._get_window()
         window = min(window, len(obj)) if not self.center else window
 
+        min_periods = self.min_periods
+        if min_periods is not None and not self.center:
+            # this is required as window is mutated above
```

Comment:
Can you clarify a bit further? I don't think we do this type of check in other functions like `cov` - just curious why we would have to do here

Reply:
> I think looks good but given scope of change I don't think this is suitable for 1.0 - it isn't backwards compatible in its current form with the change to how count works, so probably best for 1.1

@WillAyd Would you think it's a good idea if we roll-out the previous commit which is backward-compatible for V1.0.0, then another PR to roll-out this long term fix for V1.1?

I will probably further look if we can clean this part too for V1.1
```python
window = min(window, len(obj)) if not self.center else window

...

min_periods = self.min_periods
        if min_periods is not None and not self.center:
            # this is required as window is mutated above
            min_periods = min(min_periods, window)
```

Reply:
@WillAyd You are right, after rounds of code cleaning, this part `window = min(window, len(obj)) if not self.center else window` is no longer needed, so the part you commented is also not needed any more.